piptools/scripts/compile.py

```diff
@@ -477,14 +483,36 @@ def cli(
         find_links=repository.finder.find_links,
         emit_find_links=emit_find_links,
     )
-    writer.write(
-        results=results,
-        unsafe_requirements=resolver.unsafe_constraints,
-        markers={
-            key_from_ireq(ireq): ireq.markers for ireq in constraints if ireq.markers
-        },
-        hashes=hashes,
-    )
+    markers = {key_from_ireq(ireq): ireq.markers for ireq in constraints
+               if ireq.markers}
+
+    def write(writer, resolver, results, generate_hashes, output_file):
```

Comment:
since you nested this function, we may as well drop the `writer` and `resolver` args.

Reply:
right, I dropped `generate_hashes` as well in the latest commit